transport/transport.go

```diff
@@ -202,6 +215,10 @@ func (t *Transport) run(ctx context.Context) {
 				return
 			}
 
+			t.table.AddPeer(remote, addr)
```

Comment:
I had a similar issue with this but couldn't come up with a better solution (which is why I moved the peer table to transport instead). To talk back to a node who is gossiping with you, you need to add them to your table so you can send them a pull request, but a peer does not have access to the address of the node. So, it exists here instead of the gossip file

Reply:
Whoever calls `Transport.Send` should also be given access to the table, and can call `table.AddPeer` if necessary.

Reply:
Transport provides access to its table through the `Table()` method and the peer has access to the table using its underlying transport

```go
func (p *Peer) Table() dht.Table {
	return p.transport.Table()
}
```